functorch/_src/eager_transforms.py

```diff
@@ -1206,21 +1206,21 @@ def _wrap_all_tensors_to_functional(tensor_pytree, level):
     return tree_map(partial(_maybe_wrap_functional_tensor, level=level), tensor_pytree)
 
 
-def _maybe_unwrap_functional_tensor(maybe_tensor):
+def _maybe_unwrap_functional_tensor(maybe_tensor, *, reapply_views: bool):
     if not isinstance(maybe_tensor, torch.Tensor):
         return maybe_tensor
-    return _unwrap_functional_tensor(maybe_tensor)
+    return _unwrap_functional_tensor(maybe_tensor, reapply_views)
 
 
-def _unwrap_all_tensors_from_functional(tensor_pytree):
-    return tree_map(_maybe_unwrap_functional_tensor, tensor_pytree)
+def _unwrap_all_tensors_from_functional(tensor_pytree, *, reapply_views: bool):
+    return tree_map(lambda t: _maybe_unwrap_functional_tensor(t, reapply_views=reapply_views), tensor_pytree)
 
 
-def functionalize(func: Callable) -> Callable:
+def functionalize(func: Callable, *, reapply_views: bool = False) -> Callable:
```

Comment:
From a user's perspective: they don't know what it means to "reapply views" because they don't have the context that  functionalization involves (1) removing views and mutations and then (2) reapplying the views.

Would `functionalize(func, *, remove_views)` be a more descriptive name?

Reply:
Regarding the default: I think of functionalization as "remove in-place operations only", so "removing views" seems like an additional feature on top. Because of that, I think the default should be "remove in-place operations and reapply views".

Reply:
That all sounds totally reasonable to me:

(1) default is "only removing inplace"
(2) string argument instead of bool, with a nice descriptive error message